modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TreeTableRowSkinTest.java

```diff
@@ -356,6 +358,51 @@ public void cellsMustBeCollectableAfterRowFactoryChange() {
         JMemoryBuddy.assertCollectable(ref);
     }
 
+    // See JDK-8285700
+    @Test
+    public void testGraphicVisibilityOfTreeItem() {
+        TreeTableView<String> tree = new TreeTableView<>();
+        stageLoader = new StageLoader(tree);
+
+        TreeTableColumn<String, String> column = new TreeTableColumn<>("Column");
+        tree.getColumns().add(column);
+        tree.setRoot(new TreeItem("Root"));
+
+        for (int i = 0; i < 4; i++) {
+            TreeItem parent = new TreeItem("item - " + i, new Rectangle(10, 10));
+            TreeItem subNode = new TreeItem("sub-node - " + i, new Rectangle(10, 10));
+            parent.getChildren().addAll(subNode);
+            tree.getRoot().getChildren().add(parent);
+        }
+        tree.getRoot().setExpanded(true);
+
+        int itemNodeSize = tree.getRoot().getChildren().size();
+        TreeItem treeItem = (TreeItem)tree.getRoot().getChildren().get(itemNodeSize - 2);
+        TreeItem treeItemSubNode = (TreeItem)tree.getRoot().getChildren().get(itemNodeSize - 2).getChildren().get(0);
+        TreeItem lastItem = (TreeItem)tree.getRoot().getChildren().get(itemNodeSize - 1);
+
+        Toolkit.getToolkit().firePulse();
+        assertFalse(treeItem.isExpanded());
+        double lastItemBeforeExpand = lastItem.getGraphic().localToScene(lastItem.getGraphic().getBoundsInLocal()).getMinY();
+
+        treeItem.setExpanded(true);
+        Toolkit.getToolkit().firePulse();
+        assertTrue(treeItem.isExpanded());
+        assertNotEquals(lastItemBeforeExpand, lastItem.getGraphic().localToScene(lastItem.getGraphic().getBoundsInLocal()).getMinY());
+        double treeItemSubNodeAfterExpand = treeItemSubNode.getGraphic().localToScene(treeItemSubNode.getGraphic().getBoundsInLocal()).getMinY();
+
+        treeItem.setExpanded(false);
+        Toolkit.getToolkit().firePulse();
+        assertFalse(treeItem.isExpanded());
+        assertEquals(lastItemBeforeExpand, lastItem.getGraphic().localToScene(lastItem.getGraphic().getBoundsInLocal()).getMinY());
+        assertNotEquals(treeItemSubNodeAfterExpand, treeItemSubNode.getGraphic().localToScene(treeItemSubNode.getGraphic().getBoundsInLocal()).getMinY());
+
+        treeItem.setExpanded(true);
+        Toolkit.getToolkit().firePulse();
+        assertTrue(treeItem.isExpanded());
+        assertEquals(treeItemSubNodeAfterExpand, treeItemSubNode.getGraphic().localToScene(treeItemSubNode.getGraphic().getBoundsInLocal()).getMinY());
+    }
+
     @AfterEach
     public void after() {
```

Comment:
unrelated to the current PR, but could this method be more defensive, to avoid any problems with the invocation order?

```
    @AfterEach
    public void after() {
        if (stageLoader != null) {
            stageLoader.dispose();
            stageLoader = null;
        }
    }
```

Reply:
This looks like a good check to have.
Made above suggested code changes.